### PR TITLE
Fix RLP deserialzation for Enum with holes

### PR DESCRIPTION
**Pull request by kdeme**
RLP Enum deserialization would currently not check if "hole values" were attempted to be converted to the enum type. Now use checkedEnumAssign and fail with RlpTypeMismatch on invalid values.

There is at least one occurance of an enum with holes in rlpx p2p: DisconnectionReason. For this enum the issue could occur.

Also:
- Added enum RLP tests and rlpx p2p disconnect message tests to test the DisconnectionReason with enum hole value.
- Fixed worse custom DisconnectionReason decoding occurance in rlpx in waitSingleMsg proc where this issue could occur.

### CODE REVIEW

File: tests/p2p/test_rlpx_thunk.json [hunk 1]
```diff
@@ -59,15 +59,25 @@
     "error": "RlpTypeMismatch",
     "description": "listElem to assert on not a single entry list"
   },
-  "Listing single element list with entry off enum range": {
-    "payload": "01c116",
+  "DisconnectReason: single element list with entry out off enum range": {
+    "payload": "01c111",
```
Comment (kdeme) [inline tests/p2p/test_rlpx_thunk.json:63]: Note: I'm only changing these test values to test edge cases better

**Review by mjfh — APPROVED** [review]
